test/e2e/md_scale_test.go

```diff
@@ -94,14 +87,13 @@ var _ = Describe("When testing MachineDeployment scale out/in", func() {
 
 			// read the log of host agent container in backend, and write it
 			agentLogFile := fmt.Sprintf("/tmp/host-agent-%d.log", i)
-			func() {
-				f := WriteDockerLog(output, agentLogFile)
-				defer func() {
-					deferredErr := f.Close()
-					if deferredErr != nil {
-						Showf("error closing file %s:, %v", agentLogFile, deferredErr)
-					}
-				}()
```

Comment:
This change is done as part of https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/358 and the idea is to capture deferred errors inside a loop.

Reply:
The defer function is executed when leaving func at line 97, which is not right. It caused no agent_log generated for md_scale_test.

Reply:
> It caused no agent_log generated for md_scale_test

Could you elaborate a bit on this? Do you mean that this defer execution is required at the end of the `It` test ?
cc: @pshail

Reply:
Yes, Because there is a thread running backend to write docker log to file, we should close the file at the end of test.